File: 11. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_area, lo, hi;
        max_area = 0;
        lo = 0;
        hi = height.size() - 1;
        while (lo < hi){
            max_area = max(max_area, min(height[lo], height[hi]) * (hi - lo));
            if (height[lo] < height[hi])
                lo++;
            else
                hi--;
        }
        return max_area;
    }
};

File: Python/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans = 0
        start = 0
        end = len(height) - 1
        while start < end:
            ans = max(ans, min(height[start], height[end]) * (end - start))
            if height[start] < height[end]:
                start += 1
            else:
                end -= 1
        return ans
